knit/src/Knit/Tokenizer.hs

```diff
@@ -310,7 +316,7 @@ pSkip = Just <$> withSpan (Skipped <$> NonEmpty.some spaceChar) <|> pure Nothing
 pToken
   :: (KnownSpine components, AllConstrained (ComponentTokenizer components) components)
   => Tokenizer (Token components)
-pToken = try (pPunctuation <|> pToken' <|> pIdentifier) <|> pUnknown
+pToken = try (pPunctuation <|> pToken' <|> try pIdentifier <|> pIncompleteIdentifier) <|> pUnknown
```

Comment:
Makes sense, will do.

Reply:
What about `NonEmpty.sepBy1 (try pNameSection)`? That's still less backtracking than the entire name. Although the performance impact will be probably negligible, I'm mostly interested in having one parser for the names instead of two.

Reply:
Doesn't work (for the same reason, I think)

Reply:
Okay, we could probably write our own version of `sepBy` that has the right behaviour, but I don't know if it's worth it. I'd do it, but use your judgement.